### PR TITLE
Integrate Facebook, Linkedin service

### PR DESCRIPTION
**Pull request by ifndefdeadmau5**
- Enable login with facebook, linkedin
- Rearrange folder structure

### CODE REVIEW

File: android/app/src/main/java/com/yoda/MainActivity.java [hunk 1]
```diff
@@ -21,6 +21,13 @@ protected String getMainComponentName() {
   @Override
   public void onActivityResult(int requestCode, int resultCode, Intent data) {
     super.onActivityResult(requestCode, resultCode, data);
+    LISessionManager.getInstance(getApplicationContext()).onActivityResult(this, requestCode, resultCode, data); // <------ add here
```
Comment (ethkim) [inline android/app/src/main/java/com/yoda/MainActivity.java]: What does "add here" mean?

File: ios/linkedin-sdk.framework/Headers/LISDK.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                                      successBlock:^(NSString *returnState) {
+                                      }
+                                        errorBlock:^(NSError *error) {
+                                        }];
+ }
+ 
+ 2. use LISDKAPIHelper or LISDKDeeplinkHelper to make API calls or to perform deep link operations
+ 
+ */
+#include <linkedin-sdk/LISDKSessionManager.h>
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDK.h:31]: sort alphabetically

File: ios/linkedin-sdk.framework/Headers/LISDKAPIHelper.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                                error:^(LISDKAPIError *apiError) {
+                                   // do something with error
+                                }];
+]
+ }
+ 
+ */
+@interface LISDKAPIHelper : NSObject
+/**
+ access to singleton
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKAPIHelper.h:38]: Capitalize the first char of _every_ comment in this PR.

File: ios/linkedin-sdk.framework/Headers/LISDKAccessToken.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+@property (readonly,nonatomic) NSString *accessTokenValue;
+@property (readonly,nonatomic) NSDate *expiration;
+
++(instancetype)LISDKAccessTokenWithValue:(NSString*)value expiresOnMillis:(long long)expiresOnMillis;
+
+/**
+ create by passing in a serialized access token obtained by calling serializedString
+ */
++(instancetype)LISDKAccessTokenWithSerializedString:(NSString *)serString;
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKAccessToken.h:20]: rename serString to something more obvious: serializedAccessToken?

File: ios/linkedin-sdk.framework/Headers/LISDKAccessToken.h [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+-(NSString *)serializedString;
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKAccessToken.h:22]: same here. seralizedString doesn't mean anything from outside context.

File: ios/linkedin-sdk.framework/Headers/LISDKCallbackHandler.h [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ call this from application:openURL:sourceApplication:annotation: in AppDelegate to check if the callback can be handled by LinkedIn SDK.
+ */
++ (BOOL)shouldHandleUrl:(NSURL *)url;
+
+/**
+ call this from application:openURL:sourceApplication:annotation: in AppDelegate in order to properly handle the callbacks. This must be called only if shouldHandleUrl: returns YES.
+ */
++ (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation;
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKCallbackHandler.h:22]: Break line somewhere

File: ios/linkedin-sdk.framework/Headers/LISDKDeeplinkHelper.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                            success:(DeeplinkSuccessBlock)success
+                              error:(DeeplinkErrorBlock)error;
+
+- (void)viewOtherProfile:(NSString *)memberId
+               withState:(NSString *)state
+  showGoToAppStoreDialog:(BOOL)showDialog
+                 success:(DeeplinkSuccessBlock)success
+                   error:(DeeplinkErrorBlock)error;
+
++ (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation;
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKDeeplinkHelper.h:50]: Same as above.

File: ios/linkedin-sdk.framework/Headers/LISDKErrorCode.h [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+#ifndef LISDKErrorCode_h
+#define LISDKErrorCode_h
+
+#define LISDK_ERROR_API_DOMAIN @"LISDKErrorAPIDomain"
+#define LISDK_ERROR_AUTH_DOMAIN @"LISDKErrorAuthDomain"
+#define LISDK_ERROR_DEEPLINK_DOMAIN @"LISDKErrorDeepLinkDomain"
+
+typedef NS_ENUM(int,LISDKErrorCode) {
+    NONE,
+    /**
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKErrorCode.h:16]: // should be ok for inline comments.

File: ios/linkedin-sdk.framework/Headers/LISDKErrorCode.h [hunk 9]
```diff
@@ -0,0 +1,52 @@
+//
+//  LISDKErrorCode.h
+//
+//  Copyright (c) 2015 linkedin. All rights reserved.
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKErrorCode.h:4]: I'm surprised this isn't commented using /*\* **/. It's also poorly styled: change to LinkedIn.

File: ios/linkedin-sdk.framework/Headers/LISDKPermission.h [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+//
+//  Copyright (c) 2015 linkedin. All rights reserved.
+//
+
+#ifndef LISDKPermission_h
+#define LISDKPermission_h
+
+/** 
+ @file LISDKPermission.h
+ list of valid permissions
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKPermission.h]: List

File: ios/linkedin-sdk.framework/Headers/LISDKPermission.h [hunk 11]
```diff
@@ -0,0 +1,49 @@
+//
+//  LISDKPermission.h
```
Comment (ethkim) [inline ios/linkedin-sdk.framework/Headers/LISDKPermission.h:2]: Gosh, all this seem to be coming from LinkedIn SDK. This stuff is really badly written..

File: package.json [hunk 12]
```diff
@@ -2,17 +2,13 @@
   "name": "yoda",
   "version": "0.0.1",
   "private": true,
-  "babel": {
```
Comment (ethkim) [inline package.json:5]: Why was this removed?

File: src/components/App.js [hunk 13]
```diff
@@ -34,7 +38,49 @@ class App extends Component {
     }
 
     console.log(('Starting the app for platform: ' + this.state.os));
+    this.initLinkedIn();
+  }
+
+  // Initialize linkedin tokens
```
Comment (ethkim) [inline src/components/App.js]: replace linkedin to LinkedIn everywhere

File: src/components/App.js [hunk 14]
```diff
@@ -34,7 +38,49 @@ class App extends Component {
...
+  initLinkedIn() {
+    let redirecUrl = 'http://localhost';
```
Comment (ethkim) [inline src/components/App.js]: redirecUrl -> redirectUrl

and this can't be right. Why are we redirecting anything to localhost for client? Is react native running a server to transcode js files into native code? If that's the case, can we leave a comment what this localhost is doing?

File: src/components/App.js [hunk 15]
```diff
@@ -34,7 +38,49 @@ class App extends Component {
...
+
+  signinWithLinkedIn() {
+    LinkedInLogin.login();
+  }
 
+  componentWillMount() {
+    this.initEvent();
+  }
+
+  // fetching data from linkedin
```
Comment (ethkim) [inline src/components/App.js]: Fetching

File: src/components/App.js [hunk 16]
```diff
@@ -34,7 +38,49 @@ class App extends Component {
...
+    DeviceEventEmitter.addListener('linkedinLoginError', (error) => {
+      console.log('linkedinLoginError!');
+    });
+
+    DeviceEventEmitter.addListener('linkedinLogin', (data) => {
+      LinkedInLogin.setSession(data.accessToken, data.expiresOn);
+      LinkedInLogin.getProfile();
+    });
+
+    DeviceEventEmitter.addListener('linkedinGetRequest', (d) => {
```
Comment (ethkim) [inline src/components/App.js]: what is d? Let's use a better name.

File: src/components/utils/linkedin-login.js [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+    _clientSecret = null;
+    _state = null;
+    _scopes = null;
+
+    constructor() {
+    }
+
+    /**
+     * Initializes the LinkedinLogin API
+     * @param  {string} redirectUrl     [description]
```
Comment (ethkim) [inline src/components/utils/linkedin-login.js]: Where are the descriptions?

File: src/components/utils/linkedin-login.js [hunk 18]
```diff
@@ -0,0 +1,152 @@
...
+
+    /**
+     * Gets the Profile image
+     * @return {object} Returns the promise with the image
+     */
+    getProfileImages() {
+        const atoken = this._accessToken;
+
+        return new Promise((resolve, reject) => {
+            DeviceEventEmitter.addListener('linkedinGetRequest', (d) => {
```
Comment (ethkim) [inline src/components/utils/linkedin-login.js]: Rename d

File: src/components/utils/linkedin-login.js [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+                reject(error);
+            });
+
+            const picstr = 'https://api.linkedin.com/v1/people/~/picture-urls::(original)';
+            const picstrWithAuth = `${picstr}?oauth2_access_token=${atoken}&format=json`;
+
+            if (Platform.OS === 'android') {
+                RNLinkedinLogin.getRequest(picstr);
+            } else {
+                // if ios
```
Comment (ethkim) [inline src/components/utils/linkedin-login.js]: Move this comment next to  "} else {".

In fact, let's check explicitly if Platform.OS === 'ios'. The platform can be anything; if react native starts supporting Windows OS, our code will break as soon as we update react native version.

File: src/components/utils/linkedin-login.js [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+                console.log('picstrWithAuth', picstrWithAuth);
+
+                fetch(picstrWithAuth).then(function(response) {
+                    return response.json();
+                }).then((data) => {
+
```
Comment (ethkim) [inline src/components/utils/linkedin-login.js]: Remove this empty line.

File: src/components/utils/linkedin-login.js [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+            const profilestrWithAuth = `${profilestr}?oauth2_access_token=${atoken}&format=json`;
+
+            console.log(profilestrWithAuth);
+            if (Platform.OS === 'android') {
+                RNLinkedinLogin.getRequest(profilestr);
+            } else {
+                fetch(profilestrWithAuth).then(function(response) {
+                    return response.json();
+                }).then((data) => {
+
```
Comment (ethkim) [inline src/components/utils/linkedin-login.js]: Remove this line.

File: src/components/utils/linkedin-login.js [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+        return new Promise((resolve, reject) => {
+            const options = 'id,first-name,last-name,industry,email-address';
+            const profilestr = `https://api.linkedin.com/v1/people/~:(${options})`;
+            const profilestrWithAuth = `${profilestr}?oauth2_access_token=${atoken}&format=json`;
+
+            console.log(profilestrWithAuth);
+            if (Platform.OS === 'android') {
+                RNLinkedinLogin.getRequest(profilestr);
+            } else {
```
Comment (ethkim) [inline src/components/utils/linkedin-login.js]: Check for ios explicitly

**Comment by ethkim** [comment]
Thanks Seongmin! Lots of small comments. 

BTW, is there a way to limit the # of files changed for each PR request? 77 files changed for each PR is unwieldy..

File: android/app/src/main/java/com/yoda/MainActivity.java (hunk 1 above)
Comment (sv-yoda) [reply android/app/src/main/java/com/yoda/MainActivity.java]: It means nothing. I'll remove it.

File: ios/linkedin-sdk.framework/Headers/LISDKPermission.h (hunk 11 above)
Comment (ifndefdeadmau5) [reply ios/linkedin-sdk.framework/Headers/LISDKPermission.h:2]: Lol...Well, All of these were shoudn't be included. It was my wrong decision.

File: src/components/App.js (hunk 14 above)
Comment (ifndefdeadmau5) [reply src/components/App.js]: image
Above is comment from author of LinkedIn login module. I think we can assign whatever we want in that variable... It's bit weird but it seems doesn't matter. I'll leave comment about that. Thank you

File: src/components/utils/linkedin-login.js (hunk 17 above)
Comment (ifndefdeadmau5) [reply src/components/utils/linkedin-login.js]: I'll add description. This file is also from node module.

File: package.json (hunk 12 above)
Comment (ifndefdeadmau5) [reply package.json:5]: Since we separated repository from back-end, We don't need this anymore.

**Comment by ifndefdeadmau5** [comment]
LGTM
